### DISCUSSION TITLE
Define tiling pattern for workspace

### DISCUSSION THREAD

**Discussion by alan12345678** [i3/i3#4937] [Q&A] [answered]
One of the things I liked most about AwesomeWM was the way of defining the tiling 'per workspace'. Unless I've missed something in the i3 docs, it seems to be the case that a workspace is split vertical only; horizontal only; or toggle between the two.

Is there any way to add a script or simple config option that stipulates e.g. the following:

I have two windows open, in a horizontal split (a default choice). When a new window opens, I don't want a third horizontal split I want an automatic vertical split on one of the existing windows. I now have one half-workspace window, and two quarter-workspace windows. If I open a fourth window, I'd expect the larger (un-split) window to now be split such that I now have four 'quarter' sized windows.

That's a lot less keystrokes than at present having to always ensure the correct window is active, then define a new split, then open the new window.

**Comment by vincentbernat** [comment, marked as answer]
i3 does not come with layout. You need to use an IPC script for that. For example, there is https://aduros.com/blog/hacking-i3-automatic-layout/.

Also coming from Awesome, I was missing this first, but now, not anymore.
